File: pkg/dev_compiler/test/codegen/expect/syncstar_yieldstar_test.js
dart_library.library('syncstar_yieldstar_test', null, /* Imports */[
  "dart_runtime/dart",
  'dart/core',
  'expect'
], /* Lazy imports */[
], function(exports, dart, core, expect) {
  'use strict';
  let dartx = dart.dartx;
  function* bar() {
    let i = 1;
    let j = 1;
    while (true) {
      yield i;
      j = dart.notNull(i) + dart.notNull(j);
      i = dart.notNull(j) - dart.notNull(i);
    }
  }
  dart.fn(bar);
  function* foo() {
    yield* [1, 2, 3];
    yield null;
    yield* dart.as(bar(), core.Iterable);
  }
  dart.fn(foo);
  function main() {
    expect.Expect.listEquals([1, 2, 3, null, 1, 1, 2, 3, 5], dart.as(dart.dsend(dart.dsend(foo(), 'take', 9), 'toList'), core.List));
  }
  dart.fn(main);
  // Exports:
  exports.bar = bar;
  exports.foo = foo;
  exports.main = main;
});

File: tests/language_2/syncstar_yieldstar_test.dart
// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

// TODO(jmesserly): this is a copy of the language test of the same name,
// we can remove this copy when we're running against those tests.
import "package:expect/expect.dart";

bar() sync* {
  int i = 1;
  int j = 1;
  while (true) {
    yield i;
    j = i + j;
    i = j - i;
  }
}

foo() sync* {
  yield* [1, 2, 3];
  yield null;
  // TODO(jmesserly): added cast here to work around:
  // https://codereview.chromium.org/1213503002/
  yield* bar() as Iterable;
}

main() async {
  Expect.listEquals([1, 2, 3, null, 1, 1, 2, 3, 5], foo().take(9).toList());
}
